test/hotspot/jtreg/compiler/c2/irTests/TestBoolNodeGVN.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2.irTests;
+
+import compiler.lib.ir_framework.Argument;
+import compiler.lib.ir_framework.Arguments;
+import compiler.lib.ir_framework.CompilePhase;
+import compiler.lib.ir_framework.IR;
+import compiler.lib.ir_framework.IRNode;
+import compiler.lib.ir_framework.Test;
+import compiler.lib.ir_framework.TestFramework;
+
+/**
+ * @test
+ * @bug 8327381
+ * @summary Refactor boolean node tautology transformations
+ * @library /test/lib /
+ * @run main compiler.c2.irTests.TestBoolNodeGVN
+ */
+public class TestBoolNodeGVN {
+    public static void main(String[] args) {
+        TestFramework.run();
+        testCorrectness();
+    }
+
+    /**
+     * Test changing ((x & m) u<= m) or ((m & x) u<= m) to always true, same with ((x & m) u< m+1) and ((m & x) u< m+1)
+     * The test is not applicable to x86 (32bit) for not having <code>Integer.compareUnsigned</code> intrinsified.
+     */
+    @Test
+    @Arguments(values = {Argument.DEFAULT, Argument.DEFAULT})
+    @IR(failOn = IRNode.CMP_U, phase = CompilePhase.AFTER_PARSING, applyIfPlatform = {"x86", "false"})
```

Comment:
Is the 32bit x86 the only platform that does not support it? An alternative would be to enable the IR rule on all platforms that we know support the unsigned compare.

Reply:
Otherwise some less well tested platform might not support it, and then they have to fix up this rule later, having less context/understanding and might get things wrong then.

Reply:
Good point. I know x64 and aarch64 support it for sure, but it looks like RISCV also has supports for it [0][1]. I'm not able to verify this on RISCV for not having access to such a machine.

[0] https://github.com/openjdk/jdk/blob/44549b605a7aad1e3143a4058ef6504a7c04167a/src/hotspot/cpu/riscv/riscv.ad#L8800-L8817
[1] https://github.com/openjdk/jdk/commit/f9795d0d09a82cafb3e79ad8667e505c194d745b